### DISCUSSION TITLE
Child Devices Vs. Modules

### DISCUSSION THREAD

**Discussion by jdavidson2021** [netbox-community/netbox#9774] [Q&A]
We are having some difficulty understanding the different use cases / thoughts behind how child devices and modules are supposed to be used by the NetBox community.  We currently have over 29,000+ parent/child devices utilizing the device bay functionality when needed.  All of our child devices are inventory items (line cards, power supplies, etc.) that we want to track the location of and these devices are serialized.  When you remove a module from a device it appears that the module is deleted from NetBox, unlike when you dis-associate a child device from a parent device.  In this case, the child device still exists in NetBox and is just associated with the rack rather than a device.  This limitation of models makes it very difficult for us to use in its current form.

Why would we want to utilize the module functionality?  Are there plans to enhance the functionality of modules or is everything implemented that is going to be implemented for modules?  Are you trying to deprecate the parent child functionality?  We are just trying to better understand what appears to be a duplication of functionality in the tool.

**Comment by candlerb** [comment]
This has been discussed multiple times before. So let me give you the short version: line cards and power supplies etc are not independent devices with their own management addresses, so should not be modelled as "Devices" at all in Netbox.  Doing so causes a number of problems, for example you can't make a LAG between two interfaces in two different devices, and you can't see all the interfaces on a chassis switch or router in one place.

"Parent/Child" is not deprecated, but "Parent devices" are nothing more than physical enclosures - a way of subdividing some rack space so multiple devices can fit into the same rackable area.  Typical examples would be when you want to fit two half-width switches side by side in a single 1U of space, or a modular patch bay, or a blade server chassis.

It was never intended for line cards to be modelled this way.  In the old days, they would be modelled as "Inventory Items" and the interfaces added directly to the enclosing device.  These days, you can model as "Modules" which have their own interfaces.

However you're right that you cannot have a standalone Inventory Item or Module, nor move it to another Device as a holding area.  That is, you have to delete the item, and then recreate it somewhere else.  Netbox isn't an asset tracking system, so the concept of "stock" and "spares" doesn't really exist.

**Comment by teixemf** [comment]
_"(...) you have to delete the item, and then recreate it somewhere else. Netbox isn't an asset tracking system, so the concept of "stock" and "spares" doesn't really exist."_

I don't really agree with this argument for not having the feature of holding Modules because Netbox allows us to hold Devices and Racks. 
It even allow us to move Devices from one Rack to another one on any Location or Site.
It also allow us to move an entire Rack (!!!) to another Location or Site.

Moving Modules around is a real world reality. 
It makes sense to be organic.
